src/blob_detection.cpp

```diff
@@ -192,41 +229,13 @@ namespace Blob_Detection
 
     void BlobInfo::_getElongation()
     {
-        if( !_contourX.empty() && !_contourY.empty() && !_elongation.found ) {
+       if( !_contourX.empty() && !_contourY.empty() && !_elongation.found ) {
             if( _contourX.size() > 1 ) {
-                std::vector < uint32_t >::const_iterator x   = _contourX.cbegin();
-                std::vector < uint32_t >::const_iterator y   = _contourY.cbegin();
-                std::vector < uint32_t >::const_iterator end = _contourX.cend();
-
-                uint32_t maximumDistance = 0;
-
-                Point2d startPoint, endPoint;
-
-                for( ; x != (end - 1); ++x, ++y ) {
-                    std::vector < uint32_t >::const_iterator xx = x + 1;
-                    std::vector < uint32_t >::const_iterator yy = y + 1;
-
-                    for( ; xx != end; ++xx, ++yy ) {
-                        uint32_t distance = (*x - *xx) * (*x - *xx) + (*y - *yy) * (*y - *yy);
-
-                        if( maximumDistance < distance ) {
-                            maximumDistance = distance;
+            	Point2d startPoint, endPoint;
+            	_getLengthFromContour(_contourX, _contourY, _length.value, startPoint, endPoint);
 
-                            startPoint.x = *x;
-                            startPoint.y = *y;
-
-                            endPoint.x = *xx;
-                            endPoint.y = *xx;
-                        }
-                    }
-                }
-
-                const double length = sqrt( static_cast<double>(maximumDistance) );
-                const double angle  = -atan2( static_cast<double>(endPoint.y - startPoint.y),
-                                              static_cast<double>(endPoint.x - startPoint.x) );
-
-                const double _cos = cos( angle );
-                const double _sin = sin( angle );
+                const double _cos = cos( _guideAngle.value );
```

Comment:
We could actually calculate angle from startPoint and endPoint. I think we don't need angle as a member of the class.